Document/PostManager.php

```diff
@@ -39,6 +42,61 @@ public function findOneBySlug($year, $month, $day, $slug)
     }
 
     /**
+     * @param string        $permalink
+     * @param BlogInterface $blog
+     */
+    public function findOneByPermalink($permalink, BlogInterface $blog)
+    {
+        $repository = $this->getRepository();
+
+        $query = $repository->createQueryBuilder('p');
+
+        try {
+            $urlParameters = $blog->getPermalinkGenerator()->getParameters($permalink);
+
+            $parameters = array();
+
+            if (isset($urlParameters['year']) && isset($urlParameters['month']) && isset($urlParameters['day'])) {
+                $pdqp = $this->getPublicationDateQueryParts(sprintf('%d-%d-%d', $urlParameters['year'], $urlParameters['month'], $urlParameters['day']), 'day');
+
+                $parameters = array_merge($parameters, $pdqp['params']);
+
+                $query->andWhere($pdqp['query']);
+            }
+
+            if (isset($urlParameters['slug'])) {
+                $query->andWhere('p.slug = :slug');
+                $parameters['slug'] = $urlParameters['slug'];
+            }
+
+            if (isset($urlParameters['collection'])) {
+                $pcqp = $this->getPublicationCollectionQueryParts($urlParameters['collection']);
+
+                $parameters = array_merge($parameters, $pcqp['params']);
+
+                $query
+                    ->leftJoin('p.collection', 'c')
+                    ->andWhere($pcqp['query']);
+            }
+
+            if (count($parameters) == 0) {
+                return;
+            }
+
+            $query->setParameters($parameters);
+
+            $results = $query->getQuery()->getResult();
+
+            if (count($results) > 0) {
+                return $results[0];
+            }
+        } catch (\InvalidArgumentException $e) {
+        }
```

Comment:
I used the same logic like in the `Entity\PostManager`: https://github.com/sonata-project/SonataNewsBundle/blob/3.x/Entity/PostManager.php#L76

Not sure if we should throw these exceptions.